File: app/src/main/java/com/stevenfrew/beatprompter/TagFilter.java
package com.stevenfrew.beatprompter;

import java.util.ArrayList;

class TagFilter extends SongFilter
{
    TagFilter(String tag,ArrayList<SongFile> songs)
    {
        super(tag,songs,true);
    }

    public boolean equals(Object o) {
        return o instanceof TagFilter && mName.equals(((TagFilter) o).mName);
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/ui/filter/TagFilter.kt
package com.stevenfrew.beatprompter.ui.filter

import com.stevenfrew.beatprompter.cache.SongFile

class TagFilter(tag: String, songs: List<SongFile>) :
	SongFilter(tag, songs.map { it to null }, true) {
	override fun equals(other: Any?): Boolean = other is TagFilter && name == other.name
	override fun hashCode(): Int = javaClass.hashCode()
}